allennlp/common/file_utils.py

```diff
@@ -1085,20 +1085,27 @@ def get_file_extension(path: str, dot=True, lower: bool = True):
 
 
 def open_compressed(
-    filename: Union[str, PathLike], mode: str = "rt", encoding: Optional[str] = "UTF-8", **kwargs
+    filename: Union[str, PathLike],
+    compression_type: str = None,
+    mode: str = "rt",
+    encoding: Optional[str] = "UTF-8",
+    **kwargs,
 ):
     if not isinstance(filename, str):
         filename = str(filename)
     open_fn: Callable = open
 
-    if filename.endswith(".gz"):
-        import gzip
-
-        open_fn = gzip.open
-    elif filename.endswith(".bz2"):
-        import bz2
+    compression_modules = {"gz": "gzip", "bz2": "bz2", "lzma": "lzma"}
+    if not compression_type:
```

Comment:
If I pass in an empty string for `compression_type`, we will go down this path. I don't think that's what we want.

Reply:
Can I simply consider an empty string for compression_type to be equivalent to it being 'None' ?

Reply:
I think that would be surprising to a user of the library. You can just compare to `None`:
```suggestion
    if compression_type is None:
```